gnmi/gnmiclient/cache.go

```diff
@@ -23,29 +23,43 @@ import (
 	"google.golang.org/grpc/peer"
 
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
-	"github.com/openconfig/lemming/gnmi"
+)
+
+// Mode indicates the mode in which the gNMI service operates.
+type Mode string
+
+const (
+	// GNMIModeMetadataKey is the context metadata key used to specify the
+	// mode in which the gNMI server should operate.
+	GNMIModeMetadataKey = "gnmi-mode"
+	// ConfigMode indicates that the gNMI service will allow updates to
+	// intended configuration, but not operational state values.
+	ConfigMode Mode = "config"
+	// StateMode indicates that the gNMI service will allow updates to
+	// operational state, but not intended configuration values.
+	StateMode Mode = "state"
```

Comment:
Should these be owned by the client package or the server package? Move this back to the new package gnmistore to avoid the circular reference?

Reply:
moved to gnmi package